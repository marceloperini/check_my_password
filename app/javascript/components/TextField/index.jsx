import PropTypes from 'prop-types';
import React from 'react';

const TextField = (props) => {
  const { value, onChange, placeholder } = props;

  return (
    <div className="TextField">
      <input
        type="text"
        className="form-control"
        value={value}
        onChange={onChange}
        placeholder={placeholder}
      />
    </div>
  );
};

TextField.defaultProps = {
  value: '',
  placeholder: '',
};

TextField.propTypes = {
  value: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

export default TextField;
