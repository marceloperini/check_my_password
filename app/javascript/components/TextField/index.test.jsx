import React from 'react';
import renderer from 'react-test-renderer';

import TextField from '.';

describe('TextField', () => {
  const onChange = () => {};

  test('when value prop are seted', () => {
    const component = renderer.create(
      <TextField
        value="Value"
        onChange={onChange}
      />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('when placeholder prop are seted', () => {
    const component = renderer.create(
      <TextField
        onChange={onChange}
        placeholder="Placeholder..."
      />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('when value and placeholder props are not seted', () => {
    const component = renderer.create(
      <TextField
        onChange={onChange}
      />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
