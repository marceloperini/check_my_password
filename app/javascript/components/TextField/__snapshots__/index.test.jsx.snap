// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField when placeholder prop are seted 1`] = `
<div
  className="TextField"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Placeholder..."
    type="text"
    value=""
  />
</div>
`;

exports[`TextField when value and placeholder props are not seted 1`] = `
<div
  className="TextField"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder=""
    type="text"
    value=""
  />
</div>
`;

exports[`TextField when value prop are seted 1`] = `
<div
  className="TextField"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder=""
    type="text"
    value="Value"
  />
</div>
`;
